internal/querycoordv2/meta/channel_dist_manager.go

```diff
@@ -119,56 +120,28 @@ func (m *ChannelDistManager) GetShardLeadersByReplica(replica *Replica) map[stri
 	return ret
 }
 
-func (m *ChannelDistManager) GetChannelDistByReplica(replica *Replica) map[string][]int64 {
-	m.rwmutex.RLock()
-	defer m.rwmutex.RUnlock()
-
-	ret := make(map[string][]int64)
-	for _, node := range replica.GetNodes() {
-		channels := m.channels[node]
-		for _, dmc := range channels {
-			if dmc.GetCollectionID() == replica.GetCollectionID() {
-				channelName := dmc.GetChannelName()
-				_, ok := ret[channelName]
-				if !ok {
-					ret[channelName] = make([]int64, 0)
-				}
-				ret[channelName] = append(ret[channelName], node)
-			}
-		}
-	}
-	return ret
-}
-
-func (m *ChannelDistManager) GetByCollection(collectionID UniqueID) []*DmChannel {
+// return all channels in list which match all given filters
+func (m *ChannelDistManager) GetByFilter(filters ...ChannelDistFilter) []*DmChannel {
 	m.rwmutex.RLock()
 	defer m.rwmutex.RUnlock()
 
 	ret := make([]*DmChannel, 0)
 	for _, channels := range m.channels {
 		for _, channel := range channels {
-			if channel.CollectionID == collectionID {
+			allMatch := true
+			for _, fn := range filters {
+				if fn != nil && !fn(channel) {
+					allMatch = false
+				}
+			}
```

Comment:
NI: break when some filter returns false
also add a closure here is more readable